ts/features/walletV3/payment/screens/WalletPaymentPickPspScreen.tsx

```diff
@@ -56,44 +83,120 @@ const WalletPaymentPickPspScreen = () => {
+
+  const {
+    bottomSheet: sortPspBottomSheet,
+    present,
+    dismiss
+  } = useSortPspBottomSheet({
+    onSortChange: handleChangePspSorting
+  });
+
   const handlePspSelection = React.useCallback(
     (bundle: Bundle) => {
       dispatch(walletPaymentPickPsp(bundle));
     },
     [dispatch]
   );
 
-  // TODO just for testing purposes
-  React.useEffect(() => {
-    if (pot.isSome(pspListPot) && !canContinue) {
-      const pspList = pspListPot.value;
-
-      if (pspList.length > 0) {
-        handlePspSelection(pspList[0]);
-      }
-    }
-  }, [pspListPot, canContinue, handlePspSelection]);
-
   const handleContinue = () => {
     navigation.navigate(WalletPaymentRoutes.WALLET_PAYMENT_MAIN, {
       screen: WalletPaymentRoutes.WALLET_PAYMENT_CONFIRM
     });
   };
 
+  const sortButtonProps: ListItemHeader["endElement"] = React.useMemo(
+    () => ({
+      type: "buttonLink",
+      componentProps: {
+        label: I18n.t("wallet.payment.psp.pspSortButton"),
+        accessibilityLabel: I18n.t("wallet.payment.psp.pspSortButton"),
+        onPress: () => {
+          present();
+        }
+      }
+    }),
+    [present]
+  );
+
+  const isPspSelected = React.useCallback(
+    (psp: Bundle) =>
+      pipe(
+        selectedPspOption,
+        O.map(selectedPsp => selectedPsp.idBundle === psp.idBundle),
+        O.getOrElse(() => false)
+      ),
+    [selectedPspOption]
+  );
+
+  const SelectPspHeadingContent = React.useCallback(
+    () => (
+      <>
+        <H2>{I18n.t("wallet.payment.psp.title")}</H2>
+        <VSpacer size={16} />
+        <Body>{I18n.t("wallet.payment.psp.description")}</Body>
+        <Body>
+          {I18n.t("wallet.payment.psp.taxDescription")}{" "}
+          <Body weight="SemiBold">
+            {I18n.t("wallet.payment.psp.taxDescriptionBold")}
+          </Body>
+        </Body>
+        <VSpacer size={16} />
+        <ListItemHeader
+          label={I18n.t("wallet.payment.psp.pspTitle")}
+          accessibilityLabel={I18n.t("wallet.payment.psp.pspTitle")}
+          endElement={sortButtonProps}
+        />
+      </>
+    ),
+    [sortButtonProps]
+  );
+
   return (
-    <BaseScreenComponent goBack={true}>
-      <GradientScrollView
-        primaryActionProps={{
-          label: "Continua",
-          accessibilityLabel: "Continua",
-          onPress: handleContinue,
-          disabled: isLoading,
-          loading: isLoading
-        }}
-      >
-        <DebugPrettyPrint title="pspListPot" data={pspListPot} />
-      </GradientScrollView>
-    </BaseScreenComponent>
+    <GradientScrollView
+      primaryActionProps={{
+        label: I18n.t("wallet.payment.psp.continueButton"),
+        accessibilityLabel: I18n.t("wallet.payment.psp.continueButton"),
+        onPress: handleContinue,
+        disabled: isLoading || !canContinue,
+        loading: isLoading
+      }}
+    >
+      <SelectPspHeadingContent />
+      {!isLoading &&
+        pspList?.map((psp, index) => (
+          <>
```

Comment:
Each item should have an unique key. What about this?
```suggestion
          <React.Fragment key={`psp_${index}`}>
```

Reply:
This logic has been removed in favor of `RadioGroup` integration and has been addressed in the following commit: 487939547b2f2447a9e68139c288641a345a67ef